### PR TITLE
Update WebSites SDK, add Location parameter to Deleted Web Apps, fix clone slots bug

### PR DESCRIPTION
**Pull request by nking-1**
## Description

The user-facing changes in this pull request are a new Location parameter for the deleted web apps cmdlets, and an internal bug fix for cloning slots. The implementation for the deleted web apps cmdlets has also been updated to use a new location-based API. We are sharding our global endpoints to improve fault-tolerance and reliability, and as a result some APIs have been modified to require location information.

In order to make these changes, the WebSites SDK was updated. Likewise all tests have been recorded again, and some of the test code has been rewritten to be more reliable and not rely on pre-allocated resources.

## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [X] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [X] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by Nking92** [comment]
This is a large change. It's fine if it's merged in the next sprint. No need to rush this for the upcoming release.

File: src/Websites/Websites/Utilities/WebsitesClient.cs [hunk 1]
```diff
@@ -780,8 +781,20 @@ public IList<Snapshot> GetSiteSnapshots(string resourceGroupName, string webSite
             }
         }
 
-        public IList<DeletedSite> GetDeletedSites()
+        public IList<DeletedSite> GetDeletedSitesFromLocations(IEnumerable<string> locations)
         {
+            List<Task<IPage<DeletedSite>>> tasks = new List<Task<IPage<DeletedSite>>>();
+            foreach (string location in locations)
+            {
+                tasks.Add(WrappedWebsitesClient.DeletedWebApps.ListByLocationAsync(location));
```
Comment (cormacpayne) [inline src/Websites/Websites/Utilities/WebsitesClient.cs]: @Nking92 should we follow the `nextLink` returned from each of these operations, or are we only concerned about getting the first page for each location?

File: src/Websites/Websites.Test/ScenarioTests/WebAppSlotTests.ps1 [hunk 2]
```diff
@@ -131,37 +131,37 @@ function Test-StartStopRestartWebAppSlot
 		$slot = $slot | Stop-AzWebAppSlot
 
 		Assert-AreEqual "Stopped" $slot.State
-		$ping = PingWebApp $slot
+		#$ping = PingWebApp $slot
```
Comment (cormacpayne) [inline src/Websites/Websites.Test/ScenarioTests/WebAppSlotTests.ps1]: @Nking92 nit: for these lines you're commenting out, would you mind just removing them?

File: src/Websites/Websites/Utilities/WebsitesClient.cs (hunk 1 above)
Comment (Nking92) [reply src/Websites/Websites/Utilities/WebsitesClient.cs]: Good point, but internally we don't page the response, so this code should technically be ok for now. The first page will have all the sites.

**Pull request merged**